### ISSUE TITLE
master build broken for some time

### ISSUE THREAD

**Issue by mattgodbolt** [KhronosGroup/SPIRV-Tools#5014]
Hi there,

I administrate Compiler Explorer and we've been trying to build daily SPIRV tooling. It's been broken for us for a while and I wondered if someone could help us. We build with [this script](https://github.com/compiler-explorer/misc-builder/blob/main/build/build-spirv-tools.sh) and it fails with [this error](https://github.com/compiler-explorer/compiler-workflows/actions/runs/3661680852/jobs/6190110957) : 

```
  /root/SPIRV-Tools/source/opcode.cpp: In function 'int32_t spvOpcodeGeneratesType(spv::Op)':
  /root/SPIRV-Tools/source/opcode.cpp:345:19: error: 'OpTypeHitObjectNV' is not a member of 'spv::Op'
    345 |     case spv::Op::OpTypeHitObjectNV:
        |                   ^~~~~~~~~~~~~~~~~
  [ 18%] Building CXX object source/CMakeFiles/SPIRV-Tools-static.dir/val/validate_constants.cpp.o
```

Is it expected that the `master` should build?

**Comment by Goshido** [comment]
Confirm on Android as well

```txt
make: Entering directory 'D:/Development/Vulkan-ValidationLayers/build-android/SPIRV-Tools/android_test'
"[arm64-v8a] Generate language specific header for DebugInfo: headers <= grammar"
"[arm64-v8a] Generate language specific header for OpenCLDebugInfo100: headers <= grammar"
"[arm64-v8a] Generate language specific header for NonSemanticShaderDebugInfo100: headers <= grammar"
"[arm64-v8a] Generate enum<->string mapping <= grammar JSON files"
"[x86_64] Generate language specific header for DebugInfo: headers <= grammar"
"[x86_64] Generate language specific header for OpenCLDebugInfo100: headers <= grammar"
"[x86_64] Generate language specific header for NonSemanticShaderDebugInfo100: headers <= grammar"
"[x86_64] Generate enum<->string mapping <= grammar JSON files"
"[armeabi-v7a] Generate language specific header for DebugInfo: headers <= grammar"
"[armeabi-v7a] Generate language specific header for OpenCLDebugInfo100: headers <= grammar"
"[armeabi-v7a] Generate language specific header for NonSemanticShaderDebugInfo100: headers <= grammar"
"[armeabi-v7a] Generate enum<->string mapping <= grammar JSON files"
"[x86] Generate language specific header for DebugInfo: headers <= grammar"
"[x86] Generate language specific header for OpenCLDebugInfo100: headers <= grammar"
"[x86] Generate language specific header for NonSemanticShaderDebugInfo100: headers <= grammar"
"[x86] Generate enum<->string mapping <= grammar JSON files"
[arm64-v8a] Compile++      : SPIRV-Tools-opt <= aggressive_dead_code_elim_pass.cpp
In file included from ../source/opt/aggressive_dead_code_elim_pass.cpp:18:
In file included from ../source/opt/aggressive_dead_code_elim_pass.h:30:
In file included from ../source/opt/basic_block.h:29:
In file included from ../source/opt/instruction.h:33:
../source/opt/reflect.h:57:29: error: no member named 'OpTypeHitObjectNV' in 'spv::Op'
         opcode == spv::Op::OpTypeHitObjectNV;
                   ~~~~~~~~~^
1 error generated.
make: *** [D:/Programs/Android/Sdk/ndk/25.1.8937393/build/../build/core/build-binary.mk:424: D:\Development\Vulkan-ValidationLayers\build-android\SPIRV-Tools\build\app/local/arm64-v8a/objs/SPIRV-Tools-opt/source/opt/aggressive_dead_code_elim_pass.o] Error 1
make: Leaving directory 'D:/Development/Vulkan-ValidationLayers/build-android/SPIRV-Tools/android_test'
```

**Comment by dneto0** [comment]
SPIRV-Tools builds needs a newer SPIRV-Headers to build.
In this case it needs at least https://github.com/KhronosGroup/SPIRV-Headers/commit/c214f6f2d1a7253bb0e9f195c2dc5b0659dc99ef which was merged in SPIRV-Headers in Nov 16.

**Comment by dneto0** [comment]
It looks like the SPIRV-Tools code was updated to use a more recent feature in SPIRV-Headers, but the DEPS file was not updated.  Unfortunately the SPIRV-Tools CI bots didn't discover the problem, and I've filed #5014 to make the SPIRV-Tools CI bots use the DEPS file.  It's only fair.

**Comment by mattgodbolt** [comment]
Thanks @dneto0

**Comment by dneto0** [comment]
Landed the systematic fix #5031 to #5028
